### PR TITLE
Test suite fixes

### PR DESCRIPTION
**Pull request by keszybz**
A few minor issues noted while running tests unsafe.

### CODE REVIEW

**Comment by martinpitt** [comment]
i386 looks like a networking glitch in the cloud at first sight, restarted.

File: src/basic/process-util.c [hunk 1]
```diff
@@ -675,7 +675,7 @@ int getenv_for_pid(pid_t pid, const char *field, char **_value) {
                 }
                 line[i] = 0;
 
-                if (memcmp(line, field, l) == 0 && line[l] == '=') {
+                if (strneq(line, field, l) && line[l] == '=') {
```
Comment (martinpitt) [inline src/basic/process-util.c:678]: I don't have the full context here, but if the intention is to cover cases where `strlen(line) < l`, then `line[l` would still possibly be an out-of-bounds access?

File: src/test/test-ipcrm.c [hunk 2]
```diff
@@ -28,9 +28,11 @@ int main(int argc, char *argv[]) {
...
-                log_error_errno(r, "Failed to resolve \"%s\": %m", name);
-                return EXIT_FAILURE;
+                log_full_errno(r == -ESRCH ? LOG_NOTICE : LOG_ERR,
+                               r, "Failed to resolve \"%s\": %m", name);
+                return r == -ESRCH ? EXIT_TEST_SKIP : EXIT_FAILURE;
         }
 
-        return clean_ipc_by_uid(uid) < 0 ? EXIT_FAILURE : EXIT_SUCCESS;
+        r = clean_ipc_by_uid(uid);
+        return  r < 0 ? EXIT_FAILURE : EXIT_SUCCESS;
```
Comment (martinpitt) [inline src/test/test-ipcrm.c:37]: Entirely bikeshedding, but out of interest: Is that preferred code style these days?

File: src/basic/process-util.c (hunk 1 above)
Comment (poettering) [reply src/basic/process-util.c:678]: @martinpitt but in that case strneq() wouldn't have returned true, and thus the access would not take place...

File: src/test/test-ipcrm.c (hunk 2 above)
Comment (poettering) [reply src/test/test-ipcrm.c:37]: both is fine by me at least...

File: src/basic/process-util.c (hunk 1 above)
Comment (keszybz) [reply src/basic/process-util.c:678]: Exactly. If `strneq(..., l)` returns true, we know that the layout of `line` in memory is `[field[0], field[1], ..., field[l-1], x, ...]`, with at least l+1 characters initialized, where `x` could be `0` or some character. Either way, `line[l]` is safe to access.

File: src/test/test-ipcrm.c (hunk 2 above)
Comment (keszybz) [reply src/test/test-ipcrm.c:37]: It's only test code, but I think it's clearer this way.

**Comment by keszybz** [comment]
```
======================================================================
FAIL: test_no_failed (__main__.ServicesTest)
No failed units
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest.LSNNnL/build.Hby/systemd-debian/debian/tests/boot-and-services", line 60, in test_no_failed
    self.assertEqual(failed, [])
AssertionError: Lists differ: ['ifup@ens2.service loaded failed failed ifup for ens2'] != []

```

**Comment by evverx** [comment]
> AssertionError: Lists differ: ['ifup@ens2.service loaded failed failed ifup for ens2'] != []

Hm, the same error: https://github.com/systemd/systemd/pull/4670
https://objectstorage.prodstack4-5.canonical.com/v1/AUTH_77e2ada1e7a84929a74ba3b87153c0ac/autopkgtest-xenial-pitti-systemd-semaphore/xenial/i386/s/systemd-upstream/20161118_002926_7b0ef@/log.gz

Maybe, related to https://github.com/systemd/systemd/pull/4665

**Comment by rindeal** [comment]
While you're at it `test_readlink_and_make_absolute()` in `src/test/test-fs-util.c` calls `chdir(tempdir)` and `rm_rf(tempdir, ...)`, thus leaving the rest of the tests in a non-existing dir, which I don't think is intentional.

**Comment by keszybz** [comment]
Running the rest of the tests from an unlinked dir is not intentional, but also it shouldn't really matter.

**Comment by rindeal** [comment]
That's probably true on sane systems, but it broke eg. the sandbox feature of the Portage package manager on Gentoo.

**Comment by evverx** [comment]
> it broke eg. the sandbox feature of the Portage package manager on Gentoo

@rindeal , please, file a new issue
BTW: `test-util` run `chdir(temp)/rm_rf(temp)` too

**Pull request merged**